source/geometry_model/two_merged_boxes.cc

```diff
@@ -0,0 +1,523 @@
+      for (typename Triangulation<dim>::active_cell_iterator
+           cell = triangulation.begin_active();
+           cell != triangulation.end();
+           ++cell)
+        {
+
+          if (cell->face(0)->at_boundary())
+            {
+              // set the lithospheric part of the left boundary to indicator 2*dim
+              if (cell->face(0)->vertex(GeometryInfo<dim-1>::vertices_per_cell-1)[dim-1] > height_lith)
+                cell->face(0)->set_boundary_indicator (2*dim);
+              // reset the rest of the boundary to the default indicator 0
+              else
+                cell->face(0)->set_boundary_indicator (0);
+
+            }
+
+          if (cell->face(1)->at_boundary())
+            {
+              // set the lithospheric part of the right boundary to indicator 2*dim+1
+              if (cell->face(1)->vertex(GeometryInfo<dim-1>::vertices_per_cell-1)[dim-1] > height_lith)
+                cell->face(1)->set_boundary_indicator (2*dim+1);
+              // reset the rest of the boundary to the default indicator 1
+              else
+                cell->face(1)->set_boundary_indicator (1);
+            }
+
+          if (dim==3)
+            {
+              // set the lithospheric part of the front boundary to indicator 2*dim+2
+              if (cell->face(2)->at_boundary())
+                {
+                  if (cell->face(2)->vertex(GeometryInfo<dim-1>::vertices_per_cell-1)[dim-1] > height_lith)
+                    cell->face(2)->set_boundary_indicator (2*dim+2);
+                  // reset the rest of the boundary to the default indicator 2
+                  else
+                    cell->face(2)->set_boundary_indicator (2);
+                }
+
+              // set the lithospheric part of the back boundary to indicator 2*dim+3
+              if (cell->face(3)->at_boundary())
+                {
+                  if (cell->face(3)->vertex(GeometryInfo<dim-1>::vertices_per_cell-1)[dim-1] > height_lith)
+                    cell->face(3)->set_boundary_indicator (2*dim+3);
+                  // reset the rest of the boundary to the default indicator 3
+                  else
+                    cell->face(3)->set_boundary_indicator (3);
+                }
+            }
+        }
+    }
+
+    template <int dim>
+    void
+    TwoMergedBoxes<dim>::
+    create_coarse_mesh (parallel::distributed::Triangulation<dim> &total_coarse_grid) const
+    {
+      std::vector<unsigned int> lower_rep_vec(lower_repetitions, lower_repetitions+dim);
+      std::vector<unsigned int> upper_rep_vec(upper_repetitions, upper_repetitions+dim);
+
+      // initialize lower and upper coarse grid with same settings as total_coarse_grid
+      parallel::distributed::Triangulation<dim> lower_coarse_grid(total_coarse_grid.get_communicator());
+      parallel::distributed::Triangulation<dim> upper_coarse_grid(total_coarse_grid.get_communicator());
```

Comment:
It is not necessary to have the two pieces be distributed triangulations. In fact, this brings with it some unnecessary overhead. I think you could just use regular `Triangulation<dim>` objects here.